addons/website_sale/models/website.py

```diff
@@ -311,6 +311,11 @@ def sale_get_order(self, force_create=False, code=None, update_pricelist=False,
         if not request.session.get('sale_order_id'):
             request.session['sale_order_id'] = sale_order.id
 
+        # Remove unpublished products
+        sale_order.order_line = sale_order.order_line.filtered(
+            lambda ol: ol.product_id.is_published
```

Comment:
```suggestion
            lambda ol: ol.product_id.website_published
```
Basically, `is_published` should never be used (except in your tests to set the value on created product for instance).
I'll let you check the mixin, but is_published doesn't mean it is really published, website_published does.